src/core/load-fragment.c

```diff
@@ -4629,6 +4632,49 @@ int config_parse_load_credential(
                 }
         }
 
+        r = stat(q, &st);
+        if (r < 0) {
+                log_unit_error_errno(u, errno, "Couldn't stat credential at %s", q);
+                return 0;
+        }
+
+        if (st.st_mode & S_IFDIR) {
+                _cleanup_closedir_ DIR *dir = NULL;
+                struct dirent *dent;
+
+                dir = opendir(q);
+                if (!dir) {
+                        log_unit_error_errno(u, errno, "Couldn't open directory %s", q);
+                        return 0;
+                }
+
+                log_unit_info(u, "Loading credentials from directory %s", q);
+                FOREACH_DIRENT(dent, dir, return -errno) {
+                        _cleanup_free_ char *new_rvalue = NULL, *new_path = NULL;
+
+                        /* avoid infinite recursion */
+                        if (dent->d_type == DT_LNK) {
+                                log_unit_warning(u, "Skipping symlinked credential %s", dent->d_name);
+                                continue;
+                        }
+
+                        new_path = path_join(q, dent->d_name);
+                        if (!new_path)
+                                return log_oom();
+
+                        new_rvalue = strjoin(k, "_", dent->d_name, ":", new_path);
+                        if (!new_rvalue)
+                                return log_oom();
```

Comment:
nah, that's ugly. let's not generate configuration strings from configuration strings, that we thus parse ultimately twice.

Reply:
Moved the logic into `acquire_credentials`, let me know what you think.